File: src/main/java/it/contrader/automative/model/Preventivo.java
package it.contrader.automative.model;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

@Entity
@Table(name = "preventivo")
public class Preventivo implements Serializable {

	@Id
	private int id;
	
	
	@OneToOne(fetch=FetchType.LAZY)
	@JoinColumn(name = "idauto")
	private Auto auto;
	
	@ManyToOne(fetch=FetchType.LAZY)
	@JoinColumn(name = "idUtente")
	private Utente utente;
	
	@ManyToOne(fetch=FetchType.LAZY)
	@JoinColumn(name = "idOfficina")
	private Officina officina;
	
	@Column
	private Date Data;

	@Column 
	private String Dettagli;
	
	@Column
	private int Stato;
	
	@Column 
	private float Costo;
	
	@Column
	private String Risposta;
}

File: FrontAutomotiveAngular5/src/app/models/preventivo.ts
import { Officina } from "./Officina";
import { Utente } from "./Utente";
import { Auto } from "./Auto";

export class Preventivo {
	id: number;
	auto: Auto;
	utente: Utente;
	officina: Officina;
	data: string;
	dettagli: string;
	stato: number;
	costo: number;
	risposta: string;
	
    constructor(id:any,auto: Auto,utente: Utente, officina: Officina, data:string,dettagli:string,stato:any,costo:number,risposta:string){
            this.id=id;
            this.auto=auto;
            this.utente=utente;
            this.officina=officina;
            this.data=data;
            this.dettagli=dettagli;
            this.stato=stato;
            this.costo=costo;
            this.risposta=risposta;
            
        }
}